libopenage/unit/action.cpp

```diff
@@ -96,24 +96,18 @@ void UnitAction::face_towards(const coord::phys3 pos) {
 	}
 }
 
-// TODO remove (keep for testing)
-void UnitAction::damage_object(Unit &target, unsigned dmg) {
-	if (target.has_attribute(attr_type::damaged)) {
-		auto &dm = target.get_attribute<attr_type::damaged>();
-		if (dm.hp > dmg) {
-			dm.hp -= dmg;
-		}
-		else {
-			dm.hp = 0;
-		}
-	}
-}
+bool UnitAction::damage_unit(Unit &target) {
+	bool killed = false;
 
-void UnitAction::damage_object(Unit &target) {
 	if (target.has_attribute(attr_type::damaged)) {
 		auto &dm = target.get_attribute<attr_type::damaged>();
 
-		if (target.has_attribute(attr_type::armor) && this->entity->has_attribute(attr_type::attack)) {
+		// this is the damage calculation system
+
+		if (dm.hp == 0) {
+			// already killed, do nothing
```

Comment:
we should test this with a `Unit::is_dead()` function someday, we'll need it very often.